linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,7 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 					if ((td != null) && Annotations.IsMarked (td)) {
 						scope = assembly.MainModule.ImportReference (td).Scope;
 						hash.Add (td, scope);
```

Comment:
I don't think the line above your change is needed. The scope of the type definition doesn't need to be updated.